packages/cli-kit/src/plugins/tunnel.ts

```diff
@@ -1,5 +1,15 @@
+import {err, Result} from '../public/common/result.js'
 import {FanoutHookFunction, PluginReturnsForHook} from '../plugins.js'
 
+export type TunnelErrorType = 'invalid-provider' | 'tunnel-already-running' | 'wrong-credentials' | 'unknown'
+export class TunnelError extends Error {
```

Comment:
I'd inherit from [`ExtendableError`](https://github.com/Shopify/cli/blob/main/packages/cli-kit/src/error.ts#L7). Extendable error is a custom subclass of `Error` that allows doing comparisons.